### DISCUSSION TITLE
Documentation of Variables in Performance Diagram .yaml file

### DISCUSSION THREAD

**Discussion by dmwright526** [dtcenter/METplus#1739] [Plot Generation] [answered]
I am attempting to do a performance diagram for several different sensitivity experiments I have created and run through Grid Stat. I get the error with my current setup ("The number of series defined by series_val_1 is inconsistent with the number of settings required...") and was wondering if there is documentation available on the variables that are present in the default .yaml file and what the valid options are for these variables?

**Comment by bikegeek** [comment, marked as answer]
That error message is a result of a consistency check of the custom configuration file.  It is testing to see that each of the series you requested has a corresponding color, series_symbol, series line style, series line width, display on/off, series ordering (how you want your items ordered)

We have documentation on Read The Docs for the performance diagram, which discusses the default config file:
https://metplotpy.readthedocs.io/en/develop/Users_Guide/performance_diagram.html

and in the default file, we have for series_val_1:

> series_val_1:
>   // series_var_1
>   model:
>     // series_val_1
>     - GFS_0p25_G193
>   // series_var2
>   vx_mask:
>     // series_val_1
>     - NH_CMORPH_G193
>     // series_val_2
>     - SH_CMORPH_G193
>     // series_val_3
>     - TROP_CMORPH_G193
> 


*(I had to replace the comment '#' with '//' or the markup picks up the '#' and does some strange formatting)

so there are three series (permutation of each series item): 
(model, vx_mask)
1.  GFS_0p25_G193 with NH_CMORPH_G193
2. GFS_0p25_G193 with SH_CMORPH_G193
3. GFS_0p25_G193 with TROP_CMORPH_G193

and as a result, you will need three settings for:

- user_legend
- colors
- series_line_width
- series_symbols
- series_line_style
- plot_disp (to turn on/off which permutation above you wish to plot)
- series_order (allows you to reorder things without having to reorder all the settings above)


The configuration files are set up this way because the METviewer tool uses these scripts to generate the plots.  The default config file corresponds to what you would see for "default" values in the METviewer GUI.

**Comment by dmwright526** [comment]
Thank you! This was very helpful. I was able to get the figure working with this information and realizing my input dataset was not formatted correctly...
